src/main/java/io/vertx/serviceproxy/HelperUtils.java

```diff
@@ -141,4 +143,7 @@ public static JsonArray convertStackTrace(Throwable cause) {
       .collect(JsonArray::new, JsonArray::add, JsonArray::addAll);
   }
 
+  public static boolean isFuture(TypeInfo type) {
```

Comment:
This method shouldn't be there. This class is used by the generated `EBProxy`s and `EBProxyHandler`s and contains utilities only for these. This could result in a runtime error when this class is loaded and imports codegen types (which usually you don't depend upon at runtime).
I would love to have it in vertx-codegen since it's a very generic method, but why you just don't move it here: https://github.com/vert-x3/vertx-service-proxy/blob/master/src/main/java/io/vertx/serviceproxy/generator/model/ProxyModel.java#L176 and make it static (and protected, so i can reuse from the other module)

Reply:
The isFuture method is used in the Generator and not just the ProxyModel. The definition in the ProxyModel is not suitable.